### PR TITLE
Automatic package versioning based on NodeJS releases 

### PR DESCRIPTION
**Pull request by MekDrop**
This is unusual pull request, but I really would like that this would be included in the project, because this will add a possibility to use node-composer with such tools like [Dependabot](https://dependabot.com).

This pull request adds a GitHub Action that runs every 24 hours and checks if new NodeJS version is released. If so, it will release new version for node-composer with similar version number like NodeJS. For example if GitHub Action detects if there was a new NodeJS 14.15.1 released it automatically releases node-composer with v14.15.1-1 tag.

If node-composer is included in other PHP project and than newer NodeJS 14.15.2 is released. Such tools like [Dependabot](https://dependabot.com) will notify project owner about possible upgrade. If owner accepts this change, newer NodeJS version also for that project will be installed.

So basically this change adds possibility easier to upgrade to newer NodeJS when is possible.

This is how these automatically releases looks on GitHub:
image

There is one possible issue with this pull request. Master branch should not contain any dev code. Because every 24 hours automatic release will be generated. But from what I see, I don't think that could happen very often for this repo.

I really hope this change will be included. For me that would be very useful thing and I think for many others too.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 composer.phar
 /vendor/
 composer.lock
+.idea
```
Comment (mariusbuescher) [inline .gitignore:4]: I usually add these IDE-like files to my global `.gitignore` file, so I keep all editor/IDE related stuff out of the repo. Otherwise you end up adding this for every IDE people use. That is exhausting.

**Comment by mariusbuescher** [comment]
~I am not quite sure if this is the right way actually. I'd like to discuss this, because this project was only thought of as a proof of concept when I started it.~

~Anyway, my idea was to utilize some kind of API where I can get the latest node.js version on runtime.~

~I'd like to discuss this topic.~

Sorry, it is still early though. I like your idea, but I am not sure if I will continue to maintain this project. Actually this hasn't been touched in years. If you are interested, you can continue maintaining it.

**Comment by jakoch** [comment]
Hey guys!

I had the same approach (of versioning an installer like the software to install, e.g. 1.2.3-1.0.0).
This is hard to maintain and adds a lot of "versioning noise" to the installer project itself. 
Just because they release, you release. Imho this should be broken up.
If the installer code is unchanged, there is no need to (re)release.
Please do not bind "installer releases" to "releases of the software you want to install".

One way of doing this is to allow projects using your installer to set the desired version on the outside in their `composer.json':

```
 "require": { "mariusbuescher/node-composer": "^1.0" },
 
 "extra": {
   "mariusbuescher/node-composer": {
     "nodejs": "1.2.3"
   }
 }
```

**Comment by MekDrop** [comment]
@mariusbuescher I moved comment part about this project to another issue -  #4 I think it's better to separate different topics.

---

I got in my email Jens comment from here about that maybe this pull request about changing versioning is not a good idea, but I can't see where this comment gone. But I think maybe he is right. I think I should redo this pull request into two parts: one - that can autodetect NodeJS version; and another - create some new repo that keeps with NodeJS versioning. In this case all versioning will look much cleaner and there will be less noise. Probably, I could do create similar repo for yarn.

So, that why I'm closing this pull request.
